src/compiler/commandLineParser.ts

```diff
@@ -100,11 +100,12 @@ namespace ts {
                 "umd": ModuleKind.UMD,
                 "es6": ModuleKind.ES2015,
                 "es2015": ModuleKind.ES2015,
+                "es2018": ModuleKind.ES2018

```

Comment:
I would assume we would do the same thing we do for `--target esnext`. If you specifiy `--module esnext` we use all supported and spec'ed module features. Once dynamic imports is in an official spec, we move it to `--module es2018` (or whatever it becomes), but it still works with `--module esnext`.